pkg/controller/controlplane/genericactuator/actuator.go

```diff
@@ -155,6 +160,53 @@ func (a *actuator) Reconcile(
 	cp *extensionsv1alpha1.ControlPlane,
 	cluster *extensionscontroller.Cluster,
 ) (bool, error) {
+	if cp.Spec.Purpose != nil && *cp.Spec.Purpose == extensionsv1alpha1.Exposure {
+		return a.reconcileControlPlaneExposure(ctx, cp, cluster)
+	}
+	return a.reconcileControlPlane(ctx, cp, cluster)
+}
+
+func (a *actuator) reconcileControlPlaneExposure(
+	ctx context.Context,
+	cp *extensionsv1alpha1.ControlPlane,
+	cluster *extensionscontroller.Cluster,
+) (bool, error) {
+	if a.controlPlaneExposureChart == nil {
+		return false, nil
+	}
+
+	// Deploy secrets
```

Comment:
Probably it's safer to also do this if `exposureSecrets != nil`. Some actuators might not need them.

Reply:
I will fix it